src/Common/src/System/IO/RowConfigReader.cs

```diff
@@ -0,0 +1,116 @@
+        }
+
+        /// <summary>
+        /// Gets the next occurrence of the given key, from the current position of the reader,
+        /// or throws if no occurrence of the key exists in the remainder of the string.
+        /// </summary>
+        public string GetNextValue(string key)
+        {
+            string value;
+            if (!TryGetNextValue(key, out value))
+            {
+                throw new InvalidOperationException("Couldn't get next value with key " + key);
+            }
+            else
+            {
+                return value;
+            }
+        }
+
+        /// <summary>
+        /// Tries to get the next occurrence of the given key from the current position of the reader.
+        /// If successful, returns true and stores the result in 'value'. Otherwise, returns false.
+        /// </summary>
+        public bool TryGetNextValue(string key, out string value)
+        {
+            Debug.Assert(_buffer != null);
+            if (_currentIndex >= _buffer.Length)
+            {
+                value = null;
+                return false;
+            }
+
+            // First, find the key
+            int keyIndex = _buffer.IndexOf(key, _currentIndex);
+            if (keyIndex == -1)
+            {
+                value = null;
+                return false;
+            }
+
+            // Next, we will take the end of the line, and look backwards for the start of the value.
+            // NOTE: This assumes that the "value" does not have any whitespace in it, nor is there any
+            // after. This is the format of most "row-based" config files in /proc/net, etc.
+            int afterKey = keyIndex + key.Length;
+            int endOfLine = _buffer.IndexOf(Environment.NewLine, afterKey);
+            Debug.Assert(endOfLine != -1, "RowConfigReader needs a newline after the key, and one was not found.");
+
+            int valueIndex = _buffer.LastIndexOf('\t', endOfLine);
+            if (valueIndex == -1)
+            {
+                valueIndex = _buffer.LastIndexOf(' ', endOfLine); // try space as well
+            }
+            Debug.Assert(valueIndex != -1, "Key " + key + " was found, but no value on the same line.");
+            valueIndex++; // Get the first character after the whitespace.
+            value = _buffer.Substring(valueIndex, endOfLine - valueIndex); // Grab the whole value string.
+
+            _currentIndex = endOfLine + 1;
+            return true;
+        }
+
+        /// <summary>
+        /// Gets the next occurence of the key in the string, and parses it as an Int32.
+        /// Throws if the key is not found in the remainder of the string, or if the key
+        /// cannot be succesfully parsed into an Int32.
+        /// </summary>
+        /// <remarks>
+        /// This is mainly provided as a helper because most Linux config/info files
+        /// store integral data.
+        /// </remarks>
+        public int GetNextValueAsInt32(string key)
```

Comment:
With System.Diagnostics.Process, I found it made a significant perf impact to avoid creating the temporary strings in these cases and do the parsing manually.  See https://github.com/dotnet/corefx/blob/master/src/Common/src/System/IO/StringParser.cs#L108